starter_code/hangman.js

```diff
@@ -1,44 +1,141 @@
+    let indexLetter = keyCode - 65;
+    let convertedLetter = alphabet[indexLetter];
+    console.log(convertedLetter);
+
+    // .toUpperCase()
+
+    // only call the next check if this one is true
+    let isLetter = (keyCode >= 65 && keyCode <= 90); 
+
+    if (isLetter) {
+      this.checkClickedLetters(convertedLetter);
+    }
+    // 65 to 90 -> a to z
+    return (isLetter);
+  }
+
+  checkClickedLetters(key) {
+    // check if the letter is already in the array LETTERS
+    let isNotIncluded = !this.letters.includes(key);
+    if (isNotIncluded) {
+      this.letters.push(key);
+      this.addCorrectLetter(key);
+    }
+    return (isNotIncluded);
+  }
+
+  addCorrectLetter(key) {
+    let isAValidLetter = this.secretWord.includes(key);
+    // index do key e while para ver se a key ainda existe
+
+    if (isAValidLetter) {
+      this.guessedLetter += key;
+      this.checkWinner();
+    } else {
+      this.addWrongLetter(key);
+    }
+    return (isAValidLetter);
+  }
+
+  addWrongLetter(letter) {
+    this.wrongLetters += letter;
+    this.errorsLeft -= 1;
+    return this.checkGameOver();
+  }
+
+  checkGameOver() {
+    return (this.errorsLeft === 0);
+  }
+
+  checkWinner() {
+    let count = 0;
+    for (let i = 0; i < this.secretWord.length; i += 1) {
+      if (this.guessedLetter.includes(this.secretWord[i])) {
+        count += 1;
+      }
+    }
+    return (count === this.secretWord.length);
+  }
+
+}
+
+let startGame = document.getElementById('start-game-button');
+
+startGame.onclick = () => {
+  hangman = new Hangman();
+  hangman.getWord();
+  let randomWord = hangman.secretWord;
+  hangmanCanvas = new HangmanCanvas(randomWord);
+  // hangmanCanvas = new HangmanCanvas('IRONHACK');
+  hangmanCanvas.createBoard();
+  hangmanCanvas.drawHangman(1);
+  hangmanCanvas.drawHangman(2);
+  hangmanCanvas.drawHangman(3);
+  hangmanCanvas.drawHangman(4);
+  hangmanCanvas.drawHangman(5);
+  hangmanCanvas.drawHangman(6);
+  hangmanCanvas.drawHangman(7);
+  hangmanCanvas.drawHangman(8);
+  hangmanCanvas.drawHangman(9);
+  hangmanCanvas.drawHangman(10);
+  // hangmanCanvas.writeCorrectLetter();
+  // hangmanCanvas.writeWrongLetter('GIULIA', 2);
+  // hangmanCanvas.writeCorrectLetter(0);
+  // hangmanCanvas.writeCorrectLetter(1);
+  // hangmanCanvas.writeCorrectLetter(2);
+  // hangmanCanvas.writeCorrectLetter(4);
+};
 
-// class Hangman {
-//   constructor() {
-
-//   }
-
-//   getWord() {
-
-//   }
-
-//   checkIfLetter(keyCode) {
-
-//   }
-
-//   checkClickedLetters(key) {
-
-//   }
-
-//   addCorrectLetter(i) {
-
-//   }
-
-//   addWrongLetter(letter) {
-
-//   }
-
-//   checkGameOver() {
-
-//   }
-
-//   checkWinner() {
-
+document.onkeydown = (e) => {
+  // console.log(e.keyCode);
+  console.log(hangman.secretWord);
+  // hangman.checkIfLetter(e.keyCode);
+  // console.log(hangman.addCorrectLetter());
+  if (hangman.checkIfLetter(e.keyCode) && hangman.checkClickedLetters(e.) && hangman.addCorrectLetter(e.char)) {
```

Comment:
Cuidado com esse "IF".
```suggestion
  if (hangman.checkIfLetter(e.keyCode) && hangman.checkClickedLetters(e) && hangman.addCorrectLetter(e.key)) {
```